### PR TITLE
Feature/first-screen-legal-domicile-transfer

### PR DESCRIPTION
**Pull request by Rikkiabb**
## Description

First screen of children domicile transfer using application state machine and mocked data from data providers. 
Reason for mocking data layer is because it is an external dependecy on another team and we want to get feedback and keep the PR's as small as possible. I have couple of questions which I'll post as comments on this PR.

## Screenshots / Gifs
Screenshot 2021-01-11 at 22 42 23

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by Rikkiabb** [comment]
I have two questions which would be great if you could answer 🙏 

1. Is there any documentation about translations and the localization process? Mainly wondering so that work won’t start too late .

2. I see there is a todo on line 111 in `FormExternalDataProvider.tsx`, could you tell me if this todo is high on your backlog or not? And if it is not high, do you then mind if we would take it?

**Review by dabbeg — APPROVED** [review]
Nice!

**Comment by Rikkiabb** [comment]
@baering  Care for review? 🙏

**Comment by baering** [comment]
> I have two questions which would be great if you could answer 🙏

> 1. Is there any documentation about translations and the localization process? Mainly wondering so that work won’t start too late .

I don't think there is, but I'll ask around and get back to you on that

> 2. I see there is a todo on line 111 in `FormExternalDataProvider.tsx`, could you tell me if this todo is high on your backlog or not? And if it is not high, do you then mind if we would take it?

This is already implemented in a [draft PR](https://github.com/island-is/island.is/pull/2456) which I have yet to finish

File: libs/application/templates/children-domicile-transfer/src/forms/ChildrenDomicileTransferForm.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        }),
+      ],
+    }),
+    buildSection({
+      id: 'submitted',
+      title: 'Umsókn móttekin',
+      children: [
+        buildTextField({
+          id: 'children',
+          title: 'children',
```
Comment (baering) [inline libs/application/templates/children-domicile-transfer/src/forms/ChildrenDomicileTransferForm.ts:107]: I see that you've already mentioned tackling localisation in another PR but I wanted to mention that all the text fields also have the same id, which controls where into the `application.answers` they get persisted. I believe you already know and will change that in the same PR, but just wanted to throw it out there 👍

**Review by baering — APPROVED** [review]
LGTM 👍

File: libs/application/templates/children-domicile-transfer/src/forms/ChildrenDomicileTransferForm.ts (hunk 1 above)
Comment (Rikkiabb) [reply libs/application/templates/children-domicile-transfer/src/forms/ChildrenDomicileTransferForm.ts:107]: Yes this was just thrown up to get the side navigation correct, thanks for the info 😄

**Review by dabbeg — APPROVED** [review]
:1st_place_medal:

**Review by baering — APPROVED** [review]

**Review by pshomov — APPROVED** [review]
LGTM

**Review by dabbeg — APPROVED** [review]

**Pull request merged**